lang/java/avro/src/test/java/org/apache/avro/specific/TestRecordWithLogicalTypes.java

```diff
@@ -0,0 +1,697 @@
+/**
+ * Autogenerated by Avro
+ * 
+ * DO NOT EDIT DIRECTLY
+ */
+package org.apache.avro.specific;
+@SuppressWarnings("all")
+@org.apache.avro.specific.AvroGenerated
+public class TestRecordWithLogicalTypes extends org.apache.avro.specific.SpecificRecordBase implements org.apache.avro.specific.SpecificRecord {
+  private static final long serialVersionUID = -4211233492739285532L;
+  public static final org.apache.avro.Schema SCHEMA$ = new org.apache.avro.Schema.Parser().parse("{\"type\":\"record\",\"name\":\"TestRecordWithLogicalTypes\",\"namespace\":\"org.apache.avro.specific\",\"fields\":[{\"name\":\"b\",\"type\":\"boolean\"},{\"name\":\"i32\",\"type\":\"int\"},{\"name\":\"i64\",\"type\":\"long\"},{\"name\":\"f32\",\"type\":\"float\"},{\"name\":\"f64\",\"type\":\"double\"},{\"name\":\"s\",\"type\":[\"null\",\"string\"],\"default\":null},{\"name\":\"d\",\"type\":{\"type\":\"int\",\"logicalType\":\"date\"}},{\"name\":\"t\",\"type\":{\"type\":\"int\",\"logicalType\":\"time-millis\"}},{\"name\":\"ts\",\"type\":{\"type\":\"long\",\"logicalType\":\"timestamp-millis\"}}]}");
+  public static org.apache.avro.Schema getClassSchema() { return SCHEMA$; }
+  @Deprecated public boolean b;
+  @Deprecated public int i32;
+  @Deprecated public long i64;
+  @Deprecated public float f32;
+  @Deprecated public double f64;
+  @Deprecated public java.lang.CharSequence s;
+  @Deprecated public org.joda.time.LocalDate d;
```

Comment:
I understand that the choice of org.joda.time.LocalDate (in lieu of java.time.LocalDate) is dictated by the requirement to support java 7. Now, I am not sure how one writes multi-versioned java code and what other requirements and conventions avro needs to obey to, but should you see a way to accommodate java8 w/o forcing joda dependency (perhaps via this: http://www.threeten.org/threetenbp/), then perhaps you should consider doing so. Otherwise, asking for joda is a small price to pay.